### PR TITLE
Scaffolding for addon account creation

### PR DESCRIPTION
**Pull request by futa-ikeda**
-   Ticket: [ENG-5150]
-   Feature flag: n/a

## Purpose
- Create scaffolding for addon account creation/connection

## Summary of Changes
- Update node addon's `accountCreate` page to show input boxes for username/password and/or OAuth link

## Screenshot(s)
- API token example
image
- OAuth example
image
- Username/Password example
image

## QA Notes

[ENG-5150]: https://openscience.atlassian.net/browse/ENG-5150?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Mostly just stuff around credentials.

File: mirage/fixtures/external-storage-services.ts [hunk 1]
```diff
@@ -33,4 +51,16 @@ export default [
...
+        name: 'ownCloud',
+        credentialsFormat: CredentialsFormat.USERNAME_PASSWORD,
+        readOnly: false,
+        supportsCopy: true,
+        supportsUserSpecificRootFolder: true,
+        supportsFileVersioning: true,
+        supportsBulkDownload: true,
+        maxConcurrentDownloads: 1,
+        maxUploadMb: 5,
+    },
```
[the remark this answers is not in the document]
Comment (futa-ikeda) [reply mirage/fixtures/external-storage-services.ts:65]: Ahh, yes, I forgot S3 had two types of keys. Will add that one. GitLab has personal access token and a repository select and OwnCloud has username+password and a `Host URL` field. I can add those in this PR as well

File: app/models/external-storage-service.ts [hunk 2]
```diff
@@ -2,10 +2,19 @@ import { attr } from '@ember-data/model';
 
 import OsfModel from './osf-model';
 
+export enum CredentialsFormat {
+    OAUTH = 'oauth',
+    OAUTH2 = 'oauth2',
+    URL_USERNAME_PASSWORD = 'url_username_password',
+    REPO_TOKEN = 'repo_token',
+    ACCESS_SECRET_KEYS = 'access_secret_keys',
```
Comment (brianjgeiger) [inline app/models/external-storage-service.ts:10]: There is also a `username_password` type. Six total including both oauths.

File: lib/osf-components/addon/components/addons-service/account-setup-manager/component.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    get showDataverseRepoOptions() {
+        const { provider } = this.args;
+        return provider.id === 'dataverse' && provider.credentialsFormat === CredentialsFormat.REPO_TOKEN;
+    }
+
+
+    get showGitlabRepoOptions() {
+        const { provider } = this.args;
+        return provider.id === 'gitlab' && provider.credentialsFormat === CredentialsFormat.REPO_TOKEN;
+    }
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/addons-service/account-setup-manager/component.ts:55]: Hmmm. This should probably be a bit more generic. I know we have to figure out ultimately where this info is being stored, though. But maybe make repoOptions into an object that is keyed by the provider id?

File: lib/osf-components/addon/components/addons-service/user-addons-manager/component.ts [hunk 4]
```diff
@@ -59,6 +59,14 @@ export default class UserAddonManagerComponent extends Component<Args> {
...
     @tracked selectedProvider?: Provider;
     @tracked selectedAccount?: AuthorizedStorageAccountModel;
+    @tracked credentialsObject = {
+        url: '',
+        username: '',
+        password: '',
+        token: '',
+        accessKey: '',
+        secretKey: '',
+    };
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/addons-service/user-addons-manager/component.ts:69]: Is `repo` missing from here?

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Almost there. A little more cleanup and some generalization.

File: app/models/external-storage-service.ts (hunk 2 above)
Comment (futa-ikeda) [reply app/models/external-storage-service.ts:10]: Ahh, I missed that. Which one uses this type? I will add it to the mirage fixture as well.

Comment (brianjgeiger) [reply app/models/external-storage-service.ts:10]: Looks like it's Boa.

File: lib/osf-components/addon/components/addons-service/user-addons-manager/component.ts (hunk 4 above)
Comment (futa-ikeda) [reply lib/osf-components/addon/components/addons-service/user-addons-manager/component.ts:69]: Oh, yep it is! It's stored on the `account-setup-manager` right now and that's not right

**Review by brianjgeiger — APPROVED** [review]
I'm not suuuper keen on any place that says `provider === 'x'` but I think we can leave it for now and consider alternate approaches another day.

**Pull request merged**